api/v1beta1/openstackbaremetalset_types.go

```diff
@@ -36,10 +50,20 @@ type OpenStackBaremetalSetSpec struct {
 	// +kubebuilder:default={}
 	// +kubebuilder:validation:Optional
 	// BaremetalHosts - Map of hostname to control plane IP address for all nodes to provision
-	BaremetalHosts map[string]string `json:"baremetalHosts"`
+	BaremetalHosts map[string]InstanceSpec `json:"baremetalHosts"`
 	// RhelImageURL - Remote URL pointing to desired RHEL qcow2 image
 	RhelImageURL string `json:"rhelImageUrl,omitempty"`
 
+	// +kubebuilder:validation:Optional
+	// UserData holds the reference to the Secret containing the user
+	// data to be passed to the host before it boots.
+	UserData *corev1.SecretReference `json:"userData,omitmempty"`
+
+	// +kubebuilder:validation:Optional
+	// NetworkData holds the reference to the Secret containing network
+	// data to be passed to the hosts.
+	NetworkData *corev1.SecretReference `json:"networkData,omitmempty"`
```

Comment:
Should these comments mention something about them being used as defaults if nothing is provided for a particular host in `BaremetalHosts` map?

Reply:
done